### PR TITLE
Tests for raw block volumes are added

### LINKED ISSUE #174: sanity: add test cases for block volumes
/assign @bswartz

### PR DESCRIPTION
**Pull request by taaraora**
**What type of PR is 
/kind feature

**What this PR does / why we need it**:
It adds raw volumes tests.

**Which issue(s) this PR fixes**:

Fixes #174

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
csi-sanity --csi.testvolumeaccesstype=block now runs CSI tests with raw block volumes.
```

### CODE REVIEW

File: pkg/sanity/controller.go [hunk 1]
```diff
@@ -44,6 +45,22 @@ func TestVolumeSize(sc *TestContext) int64 {
 	return sc.Config.TestVolumeSize
 }
 
+func TestVolumeCapabilityWithAccessType(sc *TestContext, m csi.VolumeCapability_AccessMode_Mode) *csi.VolumeCapability {
```
Comment (jsafrane) [inline pkg/sanity/controller.go]: nit: `Test` prefix implies that it actually tests something.

File: pkg/sanity/controller.go [hunk 2]
```diff
@@ -966,8 +908,9 @@ var _ = DescribeSanity("Controller Service [Controller Server]", func(sc *TestCo
 			_, err = c.ValidateVolumeCapabilities(
 				context.Background(),
 				&csi.ValidateVolumeCapabilitiesRequest{
-					VolumeId: vol.GetVolume().GetVolumeId(),
-					Secrets:  sc.Secrets.ControllerValidateVolumeCapabilitiesSecret,
+					VolumeId:           vol.GetVolume().GetVolumeId(),
+					VolumeCapabilities: nil,
```
Comment (jsafrane) [inline pkg/sanity/controller.go]: Is this necessary?

**Comment by jsafrane** [comment]
Also, please fix your commit messages, they're not really useful. Simple "block volume support" would be enough.

File: pkg/sanity/controller.go (hunk 2 above)
Comment (taaraora) [reply pkg/sanity/controller.go]: No, it is not.
But the test case name is `should fail when no volume capabilities are provided` that is why I decided to emphasize that `VolumeCapabilities` is nil.

File: pkg/sanity/controller.go (hunk 1 above)
Comment (taaraora) [reply pkg/sanity/controller.go]: I was just following the way how getters are named in that file, e. g.  `TestVolumeSize`, `TestVolumeExpandSize`

**Comment by taaraora** [comment]
Thank you, commit message is fixed now.

File: pkg/sanity/controller.go (hunk 1 above)
Comment (jsafrane) [reply pkg/sanity/controller.go]: Oh, OK.

**Comment by jsafrane** [comment]
I tried with AWS EBS driver, it failed with various ListVolumes errors that look unrelated to block volumes

@davidz627 @msau42, can you please test it with GCE PD?

**Comment by saad-ali** [comment]
CC @davidz627 please check this out

**Comment by jsafrane** [comment]
No one has reported any issues with the tests and it works for AWS and hostpath driver.
/lgtm
/approve

@taaraora, there seems to be something wrong with the release note in the first comment. Can you please check it has required format (without '`\`')? I would even rephrase it to:
```
```release-note
csi-sanity --csi.testvolumeaccesstype=block now runs CSI tests with raw block volumes.
\```
```

**Comment by taaraora** [comment]
> No one has reported any issues with the tests and it works for AWS and hostpath driver.
> /lgtm
> /approve
> 
> @taaraora, there seems to be something wrong with the release note in the first comment. Can you please check it has required format (without '`\`')? I would even rephrase it to:
> 
> ```
> ```release-note
> csi-sanity --csi.testvolumeaccesstype=block now runs CSI tests with raw block volumes.
> \```
> ```

Thank you, I've fixed it.
